app/controllers/users_controller.rb

```diff
@@ -1,18 +1,53 @@
 class UsersController < ApplicationController
-  before_action :set_user
+  before_action :set_user, except: [:index, :new, :create]
+  after_action :verify_authorized, except: :index
+  after_action :verify_policy_scoped, only: :index
+
+  def index
+    @users = policy_scope(User)
```

Comment:
Why not? I don't see a better alternative. It is also part of the simple use case in the pundit readme https://github.com/elabs/pundit#scopes